scripts/e2e-ci.js

```diff
@@ -108,69 +91,86 @@ const extractEnv = async () => {
   console.log(`Extracting environment for ${target}...`)
 
   try {
-    const result = await pexec(CMD.EXTRACT_ENV, {
+    await exec(CMD.EXTRACT_ENV, {
       env: { ...process.env, ...ENV },
     })
-    console.log(result.stdout)
   } catch (err) {
-    console.log(err.stdout)
-    console.log(err.stderr)
-    process.exit(1)
+    console.log(err)
+    process.exit(err.code)
   }
 }
 
 const serve = () => {
   console.log(`Starting serve for ${argv.type} app in a child process...`)
   // Start static-serve
-  let child = spawn('node', CMD[`SERVE_${argv.type.toUpperCase()}`])
-  console.log(`Serving target project in a child process: ${child.pid}`)
-
-  child.stdout.on('data', (data) => {
-    console.log(`Child process ${child.pid} output: ${data}`)
+  let child = spawn('node', CMD[`SERVE_${argv.type.toUpperCase()}`], {
+    stdio: 'inherit',
   })
+  console.log(`Serving target project in a child process: ${child.pid}`)
 
-  child.stderr.on('data', (data) => {
-    console.log(`Child process ${child.pid} error: ${data}`)
-  })
+  return child
+}
 
-  child.on('close', (code) => {
-    console.log(`Child process exited with code ${code}`)
+const test = () => {
+  return new Promise((resolve, reject) => {
+    console.log(`Starting test command \n${CMD.TEST}`)
+    let successful = false
+
+    try {
+      let child = spawn(CMD.TEST, { stdio: 'pipe', shell: true })
+
+      child.stdout.on('data', (data) => {
+        // Tests can fail but the command still exits normally
+        // So to be sure that all tests where successful we check
+        // for the string 'All specs passed!' written to stdout by cypress runner.
+        console.log(`${data}`)
+        if (data.includes('All specs passed!')) {
+          successful = true
+        }
+      })
+
+      child.stderr.on('data', (data) => {
+        console.log(`${data}`)
+      })
+
+      child.on('close', (code, signal) => {
+        if (successful) {
+          resolve()
+        } else {
+          reject('Some tests failed!')
+        }
+      })
+
+      child.on('error', (err) => {
+        reject(err)
+      })
+    } catch (err) {
+      reject(err)
+    }
   })
-
-  return child
 }
 
 const main = async () => {
   let exitCode = 0
-  await build()
-  await extractEnv()
-  let serveChild = serve()
-
-  console.log(`Starting test command \n${CMD.TEST}`)
+  let serveChild
 
   try {
-    const testResult = await pexec(CMD.TEST)
-    console.log(testResult.stdout)
-    console.log(testResult.stderr)
-
-    // Tests can fail without the command exiting with code 1
-    // So to be sure that all tests where successful we check
-    // for the string 'All specs passed!'
-    if (!testResult.stdout.includes('All specs passed!')) {
```

Comment:
not needed anymore?

Reply:
Yes this is still needed, but implemented a bit differently to allow the test process to stream the output to stdout. This was moved to the `test()` function and the test for `'All specs passed!'` is still there.

Reply:
Right. Missed that.